bundle/wavefrontcollectors.wavefront.com.crd.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: wavefrontcollectors.wavefront.com
+spec:
+  group: wavefront.com
+  names:
+    kind: WavefrontCollector
+    listKind: WavefrontCollectorList
+    plural: wavefrontcollectors
+    singular: wavefrontcollector
+  scope: Namespaced
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            configName:
+              description: The name of the config map providing the configuration
+                for the collector instance. If empty, a default name of "collectorName-config"
+                is assumed.
+              type: string
+            daemon:
+              description: Whether to deploy the collector as a daemonset. False will
+                roll out as a deployment.
+              type: boolean
+            enableAutoUpgrade:
+              description: If set to true, Collector pods will be upgraded automatically
+                in case new minor upgrade version is available. For pinning Collector
+                to a specific version, you will need to set this option to false.
+                We support only minor version Auto Upgrades.
+              type: boolean
+            openshiftDefault:
```

Comment:
we might need a "openshift" flag similar to proxy here as well to tell operator that collector is running in openshift environment. Based on this we need to modify fetching the default images as well as fetch security tokens. (if openshift images end up requiring authentication to pull images.)

Reply:
Authentication to redhat registry is a prerequisite to deploy Openshift cluster and so user doesn't need to authenticate every time when deploying applications from redhat registry.   However i have added this flag to be consistent with proxy to collector as well.

Reply:
ok so if you specify "registry.access.redhat.com/wavefronthq/wavefront-proxy" as the image would it work ?

Reply:
It works with "registry.connect.redhat.com"